processing file asmFiles/branch.asm (1 of 22)
processing file asmFiles/cache_associate.asm (2 of 22)
processing file asmFiles/cache_ihit_before.asm (3 of 22)
processing file asmFiles/cache_ihit_dhit.asm (4 of 22)
processing file asmFiles/count_days.asm (5 of 22)
processing file asmFiles/fib.asm (6 of 22)
processing file asmFiles/hazard.control.asm (7 of 22)
processing file asmFiles/hazard.data.asm (8 of 22)
processing file asmFiles/hazard.structure.asm (9 of 22)
processing file asmFiles/jump.asm (10 of 22)
processing file asmFiles/mergesort.asm (11 of 22)
--- memsim.hex	2021-10-27 20:17:45.231060742 -0400
+++ memcpu.hex	2021-10-27 20:17:45.159058132 -0400
@@ -87,33 +87,33 @@
 :0400C4000000000731
 :0400C5000000000B2C
 :0400C6000000000C2A
-:0400C7000000001520
-:0400C800000000161E
-:0400C900000000161D
-:0400CA000000001919
-:0400CB000000001B16
-:0400CC000000001D13
-:0400CD00000000200F
-:0400CE00000000210D
-:0400CF000000002904
-:0400D00000000033F9
-:0400D10000000036F5
-:0400D20000000038F2
-:0400D3000000003DEC
-:0400D4000000003DEB
-:0400D5000000003EE9
-:0400D60000000042E4
-:0400D70000000047DE
-:0400D8000000004BD9
-:0400D9000000004DD6
-:0400DA0000000050D2
-:0400DB0000000051D0
-:0400DC0000000051CF
-:0400DD0000000053CC
-:0400DE000000005AC4
-:0400DF0000000063BA
-:0400E00000000064B8
-:0400E1000000000714
+:0400C700000000161F
+:0400C8000000003301
+:0400C90000000051E2
+:0400CA000000005AD8
+:0400CB000000001918
+:0400CC000000002907
+:0400CD0000000050DF
+:0400CE000000001519
+:0400CF000000001617
+:0400D0000000001B11
+:0400D1000000001615
+:0400D2000000001D0D
+:0400D3000000000722
+:0400D4000000002107
+:0400D50000000063C4
+:0400D6000000001B0B
+:0400D70000000064C1
+:0400D80000000042E2
+:0400D9000000003DE6
+:0400DA000000002002
+:0400DB000000000120
+:0400DC0000000036EA
+:0400DD00000000041B
+:0400DE000000003DE1
+:0400DF0000000038E5
+:0400E0000000003DDF
+:0400E10000000030EB
 :0400E2000000000713
 :0400E3000000000712
 :0400E4000000000810
@@ -124,91 +124,49 @@
 :0400E9000000000F04
 :0400EA000000001002
 :0400EB000000001100
-:0400EC0000000017F9
-:0400ED000000001AF5
+:0400EC0000000020F0
+:0400ED0000000020EF
 :0400EE0000000020EE
-:0400EF0000000020ED
-:0400F00000000023E9
-:0400F10000000024E7
-:0400F20000000026E4
-:0400F30000000026E3
-:0400F40000000028E0
-:0400F5000000002FD8
-:0400F60000000030D6
-:0400F70000000032D3
-:0400F80000000041C3
-:0400F90000000042C1
-:0400FA0000000043BF
-:0400FB0000000044BD
-:0400FC0000000059A7
-:0400FD000000005BA4
-:0400FE000000005CA2
-:0400FF000000005CA1
-:040100000000006497
-:0401400000000001BA
-:0401410000000003B7
-:0401420000000004B5
-:0401430000000007B1
-:0401440000000007B0
-:0401450000000007AF
-:0401460000000007AE
-:0401470000000008AC
-:0401480000000008AB
-:0401490000000008AA
-:04014A0000000009A8
-:04014B000000000AA6
-:04014C000000000BA4
-:04014D000000000CA2
-:04014E000000000F9E
-:04014F00000000109C
-:04015000000000119A
-:040151000000001595
-:040152000000001693
-:040153000000001692
-:040154000000001790
-:04015500000000198D
-:040156000000001A8B
-:040157000000001B89
-:040158000000001D86
-:040159000000002082
-:04015A000000002081
-:04015B000000002080
-:04015C00000000217E
-:04015D00000000237B
-:04015E000000002479
-:04015F000000002676
-:040160000000002675
-:040161000000002872
-:040162000000002970
-:040163000000002F69
-:040164000000003067
-:040165000000003264
-:040166000000003362
-:04016700000000365E
-:04016800000000385B
-:040169000000003D55
-:04016A000000003D54
-:04016B000000003E52
-:04016C00000000414E
-:04016D00000000424C
-:04016E00000000424B
-:04016F000000004349
-:040170000000004447
-:040171000000004743
-:040172000000004B3E
-:040173000000004D3B
-:040174000000005037
-:040175000000005135
-:040176000000005134
-:040177000000005331
-:04017800000000592A
-:040179000000005A28
-:04017A000000005B26
-:04017B000000005C24
-:04017C000000005C23
-:04017D00000000631B
-:04017E000000006419
-:04017F000000006418
-:040C40000000057437
+:0400EF000000002FDE
+:0400F0000000002FDD
+:0400F10000000030DB
+:0400F2000000000703
+:0400F30000000017F2
+:0400F4000000005CAC
+:0400F5000000005BAC
+:0400F60000000028DE
+:0400F7000000001AEB
+:0400F80000000008FC
+:0400F90000000024DF
+:0400FA0000000026DC
+:0400FB0000000008F9
+:0400FC0000000032CE
+:0400FD0000000043BC
+:0400FE0000000044BA
+:0400FF000000005BA2
+:040100000000005C9F
+:04014C0000000008A7
+:04014D0000000009A5
+:04014E000000000AA3
+:04014F000000000F9D
+:04015A00000000336E
+:04015B00000000514F
+:040166000000002174
+:040167000000005C38
+:040172000000005B2E
+:040173000000005C2C
+:040174000000006324
+:040175000000001B6B
+:040176000000006421
+:040177000000004242
+:040178000000003D46
+:040179000000002062
+:04017A000000000180
+:04017B00000000364A
+:04017C00000000047B
+:04017D000000003D41
+:04017E000000003845
+:04017F000000003D3F
+:040C400000000400AC
 :043FFE0000000500BA
 :00000001FF
processing file asmFiles/mult.asm (12 of 22)
processing file asmFiles/mult_procedure.asm (13 of 22)
processing file asmFiles/quicksort.asm (14 of 22)
processing file asmFiles/search.asm (15 of 22)
processing file asmFiles/slt.asm (16 of 22)
--- memsim.hex	2021-10-27 20:17:53.167348365 -0400
+++ memcpu.hex	2021-10-27 20:17:53.078345140 -0400
@@ -61,13 +61,9 @@
 :04003C00AFA1001858
 :04003D00FFFFFFFFC3
 :04003F0000000001BC
-:0400410000000001BA
-:0400430000000001B8
-:040044000000FFFFBA
 :0402010000000001F8
 :0402030000000001F6
 :0402050000000001F4
-:040206000000FFFFF6
 :0402440000000001B5
 :0402450000000001B4
 :040246000000FFFFB6
processing file asmFiles/test.branchjump1.asm (17 of 22)
processing file asmFiles/test.halt.asm (18 of 22)
processing file asmFiles/test.ldst1.asm (19 of 22)
processing file asmFiles/test.ldst2.asm (20 of 22)
processing file asmFiles/test.loadstore.asm (21 of 22)
processing file asmFiles/test.rtype.asm (22 of 22)
processing file asmFiles/branch.asm (1 of 22)
--- memsim.hex	2021-10-27 20:18:25.310513283 -0400
+++ memcpu.hex	2021-10-27 20:18:25.208509587 -0400
@@ -1,4 +1,3 @@
-:0400000034010000C7
 :0400010000000005F6
 :0400020000000006F4
 :0400030034040006BB
processing file asmFiles/cache_associate.asm (2 of 22)
processing file asmFiles/cache_ihit_before.asm (3 of 22)
processing file asmFiles/cache_ihit_dhit.asm (4 of 22)
processing file asmFiles/count_days.asm (5 of 22)
processing file asmFiles/fib.asm (6 of 22)
processing file asmFiles/search.asm (1 of 1)
processing file asmFiles/mult.asm (1 of 2)
processing file asmFiles/mult_procedure.asm (2 of 2)
processing file asmFiles/test.branchjump1.asm (1 of 6)
processing file asmFiles/test.halt.asm (2 of 6)
processing file asmFiles/test.ldst1.asm (3 of 6)
--- memsim.hex	2021-10-27 20:19:03.864910555 -0400
+++ memcpu.hex	2021-10-27 20:19:03.764906930 -0400
@@ -15,8 +15,6 @@
 :04003C000000733716
 :04003D000000270197
 :04003E000000133774
-:040040000000733712
 :040041000000270193
-:040042000000133770
 :040C400000000002AE
 :00000001FF
processing file asmFiles/test.ldst2.asm (4 of 6)
--- memsim.hex	2021-10-27 20:19:05.510970210 -0400
+++ memcpu.hex	2021-10-27 20:19:05.425967130 -0400
@@ -10,9 +10,7 @@
 :04000900AC450008FA
 :04000A00AC4A000CF0
 :04000B00FFFFFFFFF5
-:040200000000733750
 :0402010000002701D1
-:0402020000001337AE
 :04020300FEEDBEEF5F
 :0403C000000073378F
 :0403C1000000270110
processing file asmFiles/test.loadstore.asm (5 of 6)
--- memsim.hex	2021-10-27 20:19:07.208031715 -0400
+++ memcpu.hex	2021-10-27 20:19:07.079027039 -0400
@@ -10,9 +10,7 @@
 :04000900AC450008FA
 :04000A00AC47000CF3
 :04000B00FFFFFFFFF5
-:040020000000733732
 :0400210000002701B3
-:040022000000133790
 :04002300DEADBEEFA1
 :04003C000000733716
 :04003D000000270197
processing file asmFiles/test.rtype.asm (6 of 6)
processing file asmFiles/branch.asm (1 of 22)
processing file asmFiles/cache_associate.asm (2 of 22)
processing file asmFiles/cache_ihit_before.asm (3 of 22)
processing file asmFiles/cache_ihit_dhit.asm (4 of 22)
processing file asmFiles/count_days.asm (5 of 22)
processing file asmFiles/fib.asm (6 of 22)
processing file asmFiles/hazard.control.asm (7 of 22)
processing file asmFiles/hazard.data.asm (8 of 22)
processing file asmFiles/hazard.structure.asm (9 of 22)
processing file asmFiles/jump.asm (10 of 22)
processing file asmFiles/mergesort.asm (11 of 22)
--- memsim.hex	2021-10-27 20:24:07.159902437 -0400
+++ memcpu.hex	2021-10-27 20:24:07.019897363 -0400
@@ -87,33 +87,33 @@
 :0400C4000000000731
 :0400C5000000000B2C
 :0400C6000000000C2A
-:0400C7000000001520
-:0400C800000000161E
-:0400C900000000161D
-:0400CA000000001919
-:0400CB000000001B16
-:0400CC000000001D13
-:0400CD00000000200F
-:0400CE00000000210D
-:0400CF000000002904
-:0400D00000000033F9
-:0400D10000000036F5
-:0400D20000000038F2
-:0400D3000000003DEC
-:0400D4000000003DEB
-:0400D5000000003EE9
-:0400D60000000042E4
-:0400D70000000047DE
-:0400D8000000004BD9
-:0400D9000000004DD6
-:0400DA0000000050D2
-:0400DB0000000051D0
-:0400DC0000000051CF
-:0400DD0000000053CC
-:0400DE000000005AC4
-:0400DF0000000063BA
-:0400E00000000064B8
-:0400E1000000000714
+:0400C700000000161F
+:0400C8000000003301
+:0400C90000000051E2
+:0400CA000000005AD8
+:0400CB000000001918
+:0400CC000000002907
+:0400CD0000000050DF
+:0400CE000000001519
+:0400CF000000001617
+:0400D0000000001B11
+:0400D1000000001615
+:0400D2000000001D0D
+:0400D3000000000722
+:0400D4000000002107
+:0400D50000000063C4
+:0400D6000000001B0B
+:0400D70000000064C1
+:0400D80000000042E2
+:0400D9000000003DE6
+:0400DA000000002002
+:0400DB000000000120
+:0400DC0000000036EA
+:0400DD00000000041B
+:0400DE000000003DE1
+:0400DF0000000038E5
+:0400E0000000003DDF
+:0400E10000000030EB
 :0400E2000000000713
 :0400E3000000000712
 :0400E4000000000810
@@ -124,91 +124,49 @@
 :0400E9000000000F04
 :0400EA000000001002
 :0400EB000000001100
-:0400EC0000000017F9
-:0400ED000000001AF5
+:0400EC0000000020F0
+:0400ED0000000020EF
 :0400EE0000000020EE
-:0400EF0000000020ED
-:0400F00000000023E9
-:0400F10000000024E7
-:0400F20000000026E4
-:0400F30000000026E3
-:0400F40000000028E0
-:0400F5000000002FD8
-:0400F60000000030D6
-:0400F70000000032D3
-:0400F80000000041C3
-:0400F90000000042C1
-:0400FA0000000043BF
-:0400FB0000000044BD
-:0400FC0000000059A7
-:0400FD000000005BA4
-:0400FE000000005CA2
-:0400FF000000005CA1
-:040100000000006497
-:0401400000000001BA
-:0401410000000003B7
-:0401420000000004B5
-:0401430000000007B1
-:0401440000000007B0
-:0401450000000007AF
-:0401460000000007AE
-:0401470000000008AC
-:0401480000000008AB
-:0401490000000008AA
-:04014A0000000009A8
-:04014B000000000AA6
-:04014C000000000BA4
-:04014D000000000CA2
-:04014E000000000F9E
-:04014F00000000109C
-:04015000000000119A
-:040151000000001595
-:040152000000001693
-:040153000000001692
-:040154000000001790
-:04015500000000198D
-:040156000000001A8B
-:040157000000001B89
-:040158000000001D86
-:040159000000002082
-:04015A000000002081
-:04015B000000002080
-:04015C00000000217E
-:04015D00000000237B
-:04015E000000002479
-:04015F000000002676
-:040160000000002675
-:040161000000002872
-:040162000000002970
-:040163000000002F69
-:040164000000003067
-:040165000000003264
-:040166000000003362
-:04016700000000365E
-:04016800000000385B
-:040169000000003D55
-:04016A000000003D54
-:04016B000000003E52
-:04016C00000000414E
-:04016D00000000424C
-:04016E00000000424B
-:04016F000000004349
-:040170000000004447
-:040171000000004743
-:040172000000004B3E
-:040173000000004D3B
-:040174000000005037
-:040175000000005135
-:040176000000005134
-:040177000000005331
-:04017800000000592A
-:040179000000005A28
-:04017A000000005B26
-:04017B000000005C24
-:04017C000000005C23
-:04017D00000000631B
-:04017E000000006419
-:04017F000000006418
-:040C40000000057437
+:0400EF000000002FDE
+:0400F0000000002FDD
+:0400F10000000030DB
+:0400F2000000000703
+:0400F30000000017F2
+:0400F4000000005CAC
+:0400F5000000005BAC
+:0400F60000000028DE
+:0400F7000000001AEB
+:0400F80000000008FC
+:0400F90000000024DF
+:0400FA0000000026DC
+:0400FB0000000008F9
+:0400FC0000000032CE
+:0400FD0000000043BC
+:0400FE0000000044BA
+:0400FF000000005BA2
+:040100000000005C9F
+:04014C0000000008A7
+:04014D0000000009A5
+:04014E000000000AA3
+:04014F000000000F9D
+:04015A00000000336E
+:04015B00000000514F
+:040166000000002174
+:040167000000005C38
+:040172000000005B2E
+:040173000000005C2C
+:040174000000006324
+:040175000000001B6B
+:040176000000006421
+:040177000000004242
+:040178000000003D46
+:040179000000002062
+:04017A000000000180
+:04017B00000000364A
+:04017C00000000047B
+:04017D000000003D41
+:04017E000000003845
+:04017F000000003D3F
+:040C400000000400AC
 :043FFE0000000500BA
 :00000001FF
processing file asmFiles/mult.asm (12 of 22)
processing file asmFiles/mult_procedure.asm (13 of 22)
processing file asmFiles/quicksort.asm (14 of 22)
processing file asmFiles/search.asm (15 of 22)
processing file asmFiles/slt.asm (16 of 22)
--- memsim.hex	2021-10-27 20:24:15.931220323 -0400
+++ memcpu.hex	2021-10-27 20:24:15.816216155 -0400
@@ -61,13 +61,9 @@
 :04003C00AFA1001858
 :04003D00FFFFFFFFC3
 :04003F0000000001BC
-:0400410000000001BA
-:0400430000000001B8
-:040044000000FFFFBA
 :0402010000000001F8
 :0402030000000001F6
 :0402050000000001F4
-:040206000000FFFFF6
 :0402440000000001B5
 :0402450000000001B4
 :040246000000FFFFB6
processing file asmFiles/test.branchjump1.asm (17 of 22)
processing file asmFiles/test.halt.asm (18 of 22)
processing file asmFiles/test.ldst1.asm (19 of 22)
processing file asmFiles/test.ldst2.asm (20 of 22)
processing file asmFiles/test.loadstore.asm (21 of 22)
processing file asmFiles/test.rtype.asm (22 of 22)
processing file asmFiles/cache_associate.asm (1 of 3)
processing file asmFiles/cache_ihit_before.asm (2 of 3)
processing file asmFiles/cache_ihit_dhit.asm (3 of 3)
processing file asmFiles/mult.asm (1 of 2)
processing file asmFiles/mult_procedure.asm (2 of 2)
processing file asmFiles/search.asm (1 of 1)
processing file asmFiles/test.branchjump1.asm (1 of 6)
processing file asmFiles/test.halt.asm (2 of 6)
processing file asmFiles/test.ldst1.asm (3 of 6)
processing file asmFiles/test.ldst2.asm (4 of 6)
processing file asmFiles/test.loadstore.asm (5 of 6)
processing file asmFiles/test.rtype.asm (6 of 6)
processing file asmFiles/fib.asm (1 of 1)
processing file asmFiles/branch.asm (1 of 22)
processing file asmFiles/cache_associate.asm (2 of 22)
processing file asmFiles/cache_ihit_before.asm (3 of 22)
processing file asmFiles/cache_ihit_dhit.asm (4 of 22)
processing file asmFiles/count_days.asm (5 of 22)
processing file asmFiles/fib.asm (6 of 22)
processing file asmFiles/hazard.control.asm (7 of 22)
processing file asmFiles/hazard.data.asm (8 of 22)
processing file asmFiles/hazard.structure.asm (9 of 22)
processing file asmFiles/jump.asm (10 of 22)
processing file asmFiles/mergesort.asm (11 of 22)
--- memsim.hex	2021-10-27 20:44:34.905377561 -0400
+++ memcpu.hex	2021-10-27 20:44:34.787373284 -0400
@@ -87,33 +87,33 @@
 :0400C4000000000731
 :0400C5000000000B2C
 :0400C6000000000C2A
-:0400C7000000001520
-:0400C800000000161E
-:0400C900000000161D
-:0400CA000000001919
-:0400CB000000001B16
-:0400CC000000001D13
-:0400CD00000000200F
-:0400CE00000000210D
-:0400CF000000002904
-:0400D00000000033F9
-:0400D10000000036F5
-:0400D20000000038F2
-:0400D3000000003DEC
-:0400D4000000003DEB
-:0400D5000000003EE9
-:0400D60000000042E4
-:0400D70000000047DE
-:0400D8000000004BD9
-:0400D9000000004DD6
-:0400DA0000000050D2
-:0400DB0000000051D0
-:0400DC0000000051CF
-:0400DD0000000053CC
-:0400DE000000005AC4
-:0400DF0000000063BA
-:0400E00000000064B8
-:0400E1000000000714
+:0400C700000000161F
+:0400C8000000003301
+:0400C90000000051E2
+:0400CA000000005AD8
+:0400CB000000001918
+:0400CC000000002907
+:0400CD0000000050DF
+:0400CE000000001519
+:0400CF000000001617
+:0400D0000000001B11
+:0400D1000000001615
+:0400D2000000001D0D
+:0400D3000000000722
+:0400D4000000002107
+:0400D50000000063C4
+:0400D6000000001B0B
+:0400D70000000064C1
+:0400D80000000042E2
+:0400D9000000003DE6
+:0400DA000000002002
+:0400DB000000000120
+:0400DC0000000036EA
+:0400DD00000000041B
+:0400DE000000003DE1
+:0400DF0000000038E5
+:0400E0000000003DDF
+:0400E10000000030EB
 :0400E2000000000713
 :0400E3000000000712
 :0400E4000000000810
@@ -124,91 +124,49 @@
 :0400E9000000000F04
 :0400EA000000001002
 :0400EB000000001100
-:0400EC0000000017F9
-:0400ED000000001AF5
+:0400EC0000000020F0
+:0400ED0000000020EF
 :0400EE0000000020EE
-:0400EF0000000020ED
-:0400F00000000023E9
-:0400F10000000024E7
-:0400F20000000026E4
-:0400F30000000026E3
-:0400F40000000028E0
-:0400F5000000002FD8
-:0400F60000000030D6
-:0400F70000000032D3
-:0400F80000000041C3
-:0400F90000000042C1
-:0400FA0000000043BF
-:0400FB0000000044BD
-:0400FC0000000059A7
-:0400FD000000005BA4
-:0400FE000000005CA2
-:0400FF000000005CA1
-:040100000000006497
-:0401400000000001BA
-:0401410000000003B7
-:0401420000000004B5
-:0401430000000007B1
-:0401440000000007B0
-:0401450000000007AF
-:0401460000000007AE
-:0401470000000008AC
-:0401480000000008AB
-:0401490000000008AA
-:04014A0000000009A8
-:04014B000000000AA6
-:04014C000000000BA4
-:04014D000000000CA2
-:04014E000000000F9E
-:04014F00000000109C
-:04015000000000119A
-:040151000000001595
-:040152000000001693
-:040153000000001692
-:040154000000001790
-:04015500000000198D
-:040156000000001A8B
-:040157000000001B89
-:040158000000001D86
-:040159000000002082
-:04015A000000002081
-:04015B000000002080
-:04015C00000000217E
-:04015D00000000237B
-:04015E000000002479
-:04015F000000002676
-:040160000000002675
-:040161000000002872
-:040162000000002970
-:040163000000002F69
-:040164000000003067
-:040165000000003264
-:040166000000003362
-:04016700000000365E
-:04016800000000385B
-:040169000000003D55
-:04016A000000003D54
-:04016B000000003E52
-:04016C00000000414E
-:04016D00000000424C
-:04016E00000000424B
-:04016F000000004349
-:040170000000004447
-:040171000000004743
-:040172000000004B3E
-:040173000000004D3B
-:040174000000005037
-:040175000000005135
-:040176000000005134
-:040177000000005331
-:04017800000000592A
-:040179000000005A28
-:04017A000000005B26
-:04017B000000005C24
-:04017C000000005C23
-:04017D00000000631B
-:04017E000000006419
-:04017F000000006418
-:040C40000000057437
+:0400EF000000002FDE
+:0400F0000000002FDD
+:0400F10000000030DB
+:0400F2000000000703
+:0400F30000000017F2
+:0400F4000000005CAC
+:0400F5000000005BAC
+:0400F60000000028DE
+:0400F7000000001AEB
+:0400F80000000008FC
+:0400F90000000024DF
+:0400FA0000000026DC
+:0400FB0000000008F9
+:0400FC0000000032CE
+:0400FD0000000043BC
+:0400FE0000000044BA
+:0400FF000000005BA2
+:040100000000005C9F
+:04014C0000000008A7
+:04014D0000000009A5
+:04014E000000000AA3
+:04014F000000000F9D
+:04015A00000000336E
+:04015B00000000514F
+:040166000000002174
+:040167000000005C38
+:040172000000005B2E
+:040173000000005C2C
+:040174000000006324
+:040175000000001B6B
+:040176000000006421
+:040177000000004242
+:040178000000003D46
+:040179000000002062
+:04017A000000000180
+:04017B00000000364A
+:04017C00000000047B
+:04017D000000003D41
+:04017E000000003845
+:04017F000000003D3F
+:040C400000000400AC
 :043FFE0000000500BA
 :00000001FF
processing file asmFiles/mult.asm (12 of 22)
processing file asmFiles/mult_procedure.asm (13 of 22)
processing file asmFiles/quicksort.asm (14 of 22)
processing file asmFiles/search.asm (15 of 22)
processing file asmFiles/slt.asm (16 of 22)
--- memsim.hex	2021-10-27 20:44:43.008671229 -0400
+++ memcpu.hex	2021-10-27 20:44:42.881666626 -0400
@@ -61,13 +61,9 @@
 :04003C00AFA1001858
 :04003D00FFFFFFFFC3
 :04003F0000000001BC
-:0400410000000001BA
-:0400430000000001B8
-:040044000000FFFFBA
 :0402010000000001F8
 :0402030000000001F6
 :0402050000000001F4
-:040206000000FFFFF6
 :0402440000000001B5
 :0402450000000001B4
 :040246000000FFFFB6
processing file asmFiles/test.branchjump1.asm (17 of 22)
processing file asmFiles/test.halt.asm (18 of 22)
processing file asmFiles/test.ldst1.asm (19 of 22)
processing file asmFiles/test.ldst2.asm (20 of 22)
processing file asmFiles/test.loadstore.asm (21 of 22)
processing file asmFiles/test.rtype.asm (22 of 22)
processing file asmFiles/slt.asm (1 of 1)
processing file asmFiles/mergesort.asm (1 of 1)
--- memsim.hex	2021-10-27 21:15:51.231372475 -0400
+++ memcpu.hex	2021-10-27 21:15:51.095367546 -0400
@@ -87,32 +87,32 @@
 :0400C4000000000731
 :0400C5000000000B2C
 :0400C6000000000C2A
-:0400C7000000001520
-:0400C800000000161E
-:0400C900000000161D
-:0400CA000000001919
-:0400CB000000001B16
-:0400CC000000001D13
-:0400CD00000000200F
-:0400CE00000000210D
-:0400CF000000002904
-:0400D00000000033F9
-:0400D10000000036F5
-:0400D20000000038F2
-:0400D3000000003DEC
-:0400D4000000003DEB
-:0400D5000000003EE9
-:0400D60000000042E4
-:0400D70000000047DE
-:0400D8000000004BD9
-:0400D9000000004DD6
-:0400DA0000000050D2
-:0400DB0000000051D0
-:0400DC0000000051CF
-:0400DD0000000053CC
-:0400DE000000005AC4
-:0400DF0000000063BA
-:0400E00000000064B8
+:0400C700000000161F
+:0400C8000000003301
+:0400C90000000051E2
+:0400CA000000005AD8
+:0400CB000000001918
+:0400CC000000002907
+:0400CD0000000050DF
+:0400CE000000001519
+:0400CF000000001617
+:0400D0000000001B11
+:0400D1000000003EED
+:0400D20000000047E3
+:0400D3000000004BDE
+:0400D4000000004DDB
+:0400D50000000051D6
+:0400D60000000053D3
+:0400D7000000001D08
+:0400D8000000002004
+:0400D9000000002102
+:0400DA000000003DE5
+:0400DB0000000042DF
+:0400DC0000000063BD
+:0400DD0000000064BB
+:0400DE0000000036E8
+:0400DF0000000038E5
+:0400E0000000003DDF
 :0400E1000000000714
 :0400E2000000000713
 :0400E3000000000712
@@ -128,23 +128,23 @@
 :0400ED000000001AF5
 :0400EE0000000020EE
 :0400EF0000000020ED
-:0400F00000000023E9
-:0400F10000000024E7
-:0400F20000000026E4
-:0400F30000000026E3
-:0400F40000000028E0
-:0400F5000000002FD8
-:0400F60000000030D6
-:0400F70000000032D3
-:0400F80000000041C3
-:0400F90000000042C1
-:0400FA0000000043BF
-:0400FB0000000044BD
-:0400FC0000000059A7
-:0400FD000000005BA4
-:0400FE000000005CA2
-:0400FF000000005CA1
-:040100000000006497
+:0400F0000000002FDD
+:0400F10000000030DB
+:0400F20000000041C9
+:0400F30000000042C7
+:0400F40000000059AF
+:0400F5000000005CAB
+:0400F60000000064A2
+:0400F70000000023E2
+:0400F80000000024E0
+:0400F90000000026DD
+:0400FA0000000026DC
+:0400FB0000000028D9
+:0400FC0000000032CE
+:0400FD0000000043BC
+:0400FE0000000044BA
+:0400FF000000005BA2
+:040100000000005C9F
 :0401400000000001BA
 :0401410000000003B7
 :0401420000000004B5
@@ -162,53 +162,53 @@
 :04014E000000000F9E
 :04014F00000000109C
 :04015000000000119A
-:040151000000001595
-:040152000000001693
-:040153000000001692
-:040154000000001790
-:04015500000000198D
-:040156000000001A8B
-:040157000000001B89
-:040158000000001D86
-:040159000000002082
-:04015A000000002081
-:04015B000000002080
-:04015C00000000217E
-:04015D00000000237B
-:04015E000000002479
-:04015F000000002676
-:040160000000002675
-:040161000000002872
-:040162000000002970
-:040163000000002F69
-:040164000000003067
-:040165000000003264
-:040166000000003362
-:04016700000000365E
-:04016800000000385B
-:040169000000003D55
-:04016A000000003D54
-:04016B000000003E52
-:04016C00000000414E
-:04016D00000000424C
+:040151000000001694
+:040152000000001792
+:040153000000001A8E
+:040154000000002087
+:040155000000002086
+:040156000000002F76
+:040157000000003074
+:040158000000003370
+:040159000000004161
+:04015A00000000425F
+:04015B00000000514F
+:04015C000000005946
+:04015D000000005A44
+:04015E000000001984
+:04015F000000002973
+:04016000000000504B
+:040161000000001585
+:040162000000001683
+:040163000000001B7D
+:040164000000003E59
+:04016500000000474F
+:040166000000004B4A
+:040167000000004D47
+:040168000000005142
+:04016900000000533F
+:04016A000000001D74
+:04016B000000002070
+:04016C00000000216E
+:04016D000000003D51
 :04016E00000000424B
-:04016F000000004349
-:040170000000004447
-:040171000000004743
-:040172000000004B3E
-:040173000000004D3B
-:040174000000005037
-:040175000000005135
-:040176000000005134
-:040177000000005331
-:04017800000000592A
-:040179000000005A28
+:04016F000000005C30
+:040170000000006328
+:040171000000006426
+:040172000000002366
+:040173000000002464
+:040174000000002661
+:040175000000002660
+:04017600000000285D
+:040177000000003252
+:040178000000004340
+:04017900000000443E
 :04017A000000005B26
 :04017B000000005C24
-:04017C000000005C23
-:04017D00000000631B
-:04017E000000006419
-:04017F000000006418
-:040C40000000057437
+:04017C00000000641B
+:04017D000000003648
+:04017E000000003845
+:04017F000000003D3F
+:040C40000000040BA1
 :043FFE0000000500BA
 :00000001FF
processing file asmFiles/branch.asm (1 of 22)
processing file asmFiles/cache_associate.asm (2 of 22)
processing file asmFiles/cache_ihit_before.asm (3 of 22)
processing file asmFiles/cache_ihit_dhit.asm (4 of 22)
processing file asmFiles/count_days.asm (5 of 22)
processing file asmFiles/fib.asm (6 of 22)
processing file asmFiles/hazard.control.asm (7 of 22)
processing file asmFiles/hazard.data.asm (8 of 22)
processing file asmFiles/hazard.structure.asm (9 of 22)
processing file asmFiles/jump.asm (10 of 22)
processing file asmFiles/mergesort.asm (11 of 22)
--- memsim.hex	2021-10-27 21:16:37.822060946 -0400
+++ memcpu.hex	2021-10-27 21:16:37.709056850 -0400
@@ -87,32 +87,32 @@
 :0400C4000000000731
 :0400C5000000000B2C
 :0400C6000000000C2A
-:0400C7000000001520
-:0400C800000000161E
-:0400C900000000161D
-:0400CA000000001919
-:0400CB000000001B16
-:0400CC000000001D13
-:0400CD00000000200F
-:0400CE00000000210D
-:0400CF000000002904
-:0400D00000000033F9
-:0400D10000000036F5
-:0400D20000000038F2
-:0400D3000000003DEC
-:0400D4000000003DEB
-:0400D5000000003EE9
-:0400D60000000042E4
-:0400D70000000047DE
-:0400D8000000004BD9
-:0400D9000000004DD6
-:0400DA0000000050D2
-:0400DB0000000051D0
-:0400DC0000000051CF
-:0400DD0000000053CC
-:0400DE000000005AC4
-:0400DF0000000063BA
-:0400E00000000064B8
+:0400C700000000161F
+:0400C8000000003301
+:0400C90000000051E2
+:0400CA000000005AD8
+:0400CB000000001918
+:0400CC000000002907
+:0400CD0000000050DF
+:0400CE000000001519
+:0400CF000000001617
+:0400D0000000001B11
+:0400D1000000003EED
+:0400D20000000047E3
+:0400D3000000004BDE
+:0400D4000000004DDB
+:0400D50000000051D6
+:0400D60000000053D3
+:0400D7000000001D08
+:0400D8000000002004
+:0400D9000000002102
+:0400DA000000003DE5
+:0400DB0000000042DF
+:0400DC0000000063BD
+:0400DD0000000064BB
+:0400DE0000000036E8
+:0400DF0000000038E5
+:0400E0000000003DDF
 :0400E1000000000714
 :0400E2000000000713
 :0400E3000000000712
@@ -128,23 +128,23 @@
 :0400ED000000001AF5
 :0400EE0000000020EE
 :0400EF0000000020ED
-:0400F00000000023E9
-:0400F10000000024E7
-:0400F20000000026E4
-:0400F30000000026E3
-:0400F40000000028E0
-:0400F5000000002FD8
-:0400F60000000030D6
-:0400F70000000032D3
-:0400F80000000041C3
-:0400F90000000042C1
-:0400FA0000000043BF
-:0400FB0000000044BD
-:0400FC0000000059A7
-:0400FD000000005BA4
-:0400FE000000005CA2
-:0400FF000000005CA1
-:040100000000006497
+:0400F0000000002FDD
+:0400F10000000030DB
+:0400F20000000041C9
+:0400F30000000042C7
+:0400F40000000059AF
+:0400F5000000005CAB
+:0400F60000000064A2
+:0400F70000000023E2
+:0400F80000000024E0
+:0400F90000000026DD
+:0400FA0000000026DC
+:0400FB0000000028D9
+:0400FC0000000032CE
+:0400FD0000000043BC
+:0400FE0000000044BA
+:0400FF000000005BA2
+:040100000000005C9F
 :0401400000000001BA
 :0401410000000003B7
 :0401420000000004B5
@@ -162,53 +162,53 @@
 :04014E000000000F9E
 :04014F00000000109C
 :04015000000000119A
-:040151000000001595
-:040152000000001693
-:040153000000001692
-:040154000000001790
-:04015500000000198D
-:040156000000001A8B
-:040157000000001B89
-:040158000000001D86
-:040159000000002082
-:04015A000000002081
-:04015B000000002080
-:04015C00000000217E
-:04015D00000000237B
-:04015E000000002479
-:04015F000000002676
-:040160000000002675
-:040161000000002872
-:040162000000002970
-:040163000000002F69
-:040164000000003067
-:040165000000003264
-:040166000000003362
-:04016700000000365E
-:04016800000000385B
-:040169000000003D55
-:04016A000000003D54
-:04016B000000003E52
-:04016C00000000414E
-:04016D00000000424C
+:040151000000001694
+:040152000000001792
+:040153000000001A8E
+:040154000000002087
+:040155000000002086
+:040156000000002F76
+:040157000000003074
+:040158000000003370
+:040159000000004161
+:04015A00000000425F
+:04015B00000000514F
+:04015C000000005946
+:04015D000000005A44
+:04015E000000001984
+:04015F000000002973
+:04016000000000504B
+:040161000000001585
+:040162000000001683
+:040163000000001B7D
+:040164000000003E59
+:04016500000000474F
+:040166000000004B4A
+:040167000000004D47
+:040168000000005142
+:04016900000000533F
+:04016A000000001D74
+:04016B000000002070
+:04016C00000000216E
+:04016D000000003D51
 :04016E00000000424B
-:04016F000000004349
-:040170000000004447
-:040171000000004743
-:040172000000004B3E
-:040173000000004D3B
-:040174000000005037
-:040175000000005135
-:040176000000005134
-:040177000000005331
-:04017800000000592A
-:040179000000005A28
+:04016F000000005C30
+:040170000000006328
+:040171000000006426
+:040172000000002366
+:040173000000002464
+:040174000000002661
+:040175000000002660
+:04017600000000285D
+:040177000000003252
+:040178000000004340
+:04017900000000443E
 :04017A000000005B26
 :04017B000000005C24
-:04017C000000005C23
-:04017D00000000631B
-:04017E000000006419
-:04017F000000006418
-:040C40000000057437
+:04017C00000000641B
+:04017D000000003648
+:04017E000000003845
+:04017F000000003D3F
+:040C40000000040BA1
 :043FFE0000000500BA
 :00000001FF
processing file asmFiles/mult.asm (12 of 22)
processing file asmFiles/mult_procedure.asm (13 of 22)
processing file asmFiles/quicksort.asm (14 of 22)
processing file asmFiles/search.asm (15 of 22)
processing file asmFiles/slt.asm (16 of 22)
processing file asmFiles/test.branchjump1.asm (17 of 22)
processing file asmFiles/test.halt.asm (18 of 22)
processing file asmFiles/test.ldst1.asm (19 of 22)
processing file asmFiles/test.ldst2.asm (20 of 22)
processing file asmFiles/test.loadstore.asm (21 of 22)
processing file asmFiles/test.rtype.asm (22 of 22)
processing file asmFiles/branch.asm (1 of 22)
processing file asmFiles/cache_associate.asm (2 of 22)
processing file asmFiles/cache_ihit_before.asm (3 of 22)
processing file asmFiles/cache_ihit_dhit.asm (4 of 22)
processing file asmFiles/count_days.asm (5 of 22)
processing file asmFiles/fib.asm (6 of 22)
processing file asmFiles/hazard.control.asm (7 of 22)
processing file asmFiles/hazard.data.asm (8 of 22)
processing file asmFiles/hazard.structure.asm (9 of 22)
processing file asmFiles/jump.asm (10 of 22)
processing file asmFiles/mergesort.asm (11 of 22)
--- memsim.hex	2021-10-27 21:19:04.045360154 -0400
+++ memcpu.hex	2021-10-27 21:19:03.959357037 -0400
@@ -87,32 +87,32 @@
 :0400C4000000000731
 :0400C5000000000B2C
 :0400C6000000000C2A
-:0400C7000000001520
-:0400C800000000161E
-:0400C900000000161D
-:0400CA000000001919
-:0400CB000000001B16
-:0400CC000000001D13
-:0400CD00000000200F
-:0400CE00000000210D
-:0400CF000000002904
-:0400D00000000033F9
-:0400D10000000036F5
-:0400D20000000038F2
-:0400D3000000003DEC
-:0400D4000000003DEB
-:0400D5000000003EE9
-:0400D60000000042E4
-:0400D70000000047DE
-:0400D8000000004BD9
-:0400D9000000004DD6
-:0400DA0000000050D2
-:0400DB0000000051D0
-:0400DC0000000051CF
-:0400DD0000000053CC
-:0400DE000000005AC4
-:0400DF0000000063BA
-:0400E00000000064B8
+:0400C700000000161F
+:0400C8000000003301
+:0400C90000000051E2
+:0400CA000000005AD8
+:0400CB000000001918
+:0400CC000000002907
+:0400CD0000000050DF
+:0400CE000000001519
+:0400CF000000001617
+:0400D0000000001B11
+:0400D1000000003EED
+:0400D20000000047E3
+:0400D3000000004BDE
+:0400D4000000004DDB
+:0400D50000000051D6
+:0400D60000000053D3
+:0400D7000000001D08
+:0400D8000000002004
+:0400D9000000002102
+:0400DA000000003DE5
+:0400DB0000000042DF
+:0400DC0000000063BD
+:0400DD0000000064BB
+:0400DE0000000036E8
+:0400DF0000000038E5
+:0400E0000000003DDF
 :0400E1000000000714
 :0400E2000000000713
 :0400E3000000000712
@@ -128,23 +128,23 @@
 :0400ED000000001AF5
 :0400EE0000000020EE
 :0400EF0000000020ED
-:0400F00000000023E9
-:0400F10000000024E7
-:0400F20000000026E4
-:0400F30000000026E3
-:0400F40000000028E0
-:0400F5000000002FD8
-:0400F60000000030D6
-:0400F70000000032D3
-:0400F80000000041C3
-:0400F90000000042C1
-:0400FA0000000043BF
-:0400FB0000000044BD
-:0400FC0000000059A7
-:0400FD000000005BA4
-:0400FE000000005CA2
-:0400FF000000005CA1
-:040100000000006497
+:0400F0000000002FDD
+:0400F10000000030DB
+:0400F20000000041C9
+:0400F30000000042C7
+:0400F40000000059AF
+:0400F5000000005CAB
+:0400F60000000064A2
+:0400F70000000023E2
+:0400F80000000024E0
+:0400F90000000026DD
+:0400FA0000000026DC
+:0400FB0000000028D9
+:0400FC0000000032CE
+:0400FD0000000043BC
+:0400FE0000000044BA
+:0400FF000000005BA2
+:040100000000005C9F
 :0401400000000001BA
 :0401410000000003B7
 :0401420000000004B5
@@ -162,53 +162,53 @@
 :04014E000000000F9E
 :04014F00000000109C
 :04015000000000119A
-:040151000000001595
-:040152000000001693
-:040153000000001692
-:040154000000001790
-:04015500000000198D
-:040156000000001A8B
-:040157000000001B89
-:040158000000001D86
-:040159000000002082
-:04015A000000002081
-:04015B000000002080
-:04015C00000000217E
-:04015D00000000237B
-:04015E000000002479
-:04015F000000002676
-:040160000000002675
-:040161000000002872
-:040162000000002970
-:040163000000002F69
-:040164000000003067
-:040165000000003264
-:040166000000003362
-:04016700000000365E
-:04016800000000385B
-:040169000000003D55
-:04016A000000003D54
-:04016B000000003E52
-:04016C00000000414E
-:04016D00000000424C
+:040151000000001694
+:040152000000001792
+:040153000000001A8E
+:040154000000002087
+:040155000000002086
+:040156000000002F76
+:040157000000003074
+:040158000000003370
+:040159000000004161
+:04015A00000000425F
+:04015B00000000514F
+:04015C000000005946
+:04015D000000005A44
+:04015E000000001984
+:04015F000000002973
+:04016000000000504B
+:040161000000001585
+:040162000000001683
+:040163000000001B7D
+:040164000000003E59
+:04016500000000474F
+:040166000000004B4A
+:040167000000004D47
+:040168000000005142
+:04016900000000533F
+:04016A000000001D74
+:04016B000000002070
+:04016C00000000216E
+:04016D000000003D51
 :04016E00000000424B
-:04016F000000004349
-:040170000000004447
-:040171000000004743
-:040172000000004B3E
-:040173000000004D3B
-:040174000000005037
-:040175000000005135
-:040176000000005134
-:040177000000005331
-:04017800000000592A
-:040179000000005A28
+:04016F000000005C30
+:040170000000006328
+:040171000000006426
+:040172000000002366
+:040173000000002464
+:040174000000002661
+:040175000000002660
+:04017600000000285D
+:040177000000003252
+:040178000000004340
+:04017900000000443E
 :04017A000000005B26
 :04017B000000005C24
-:04017C000000005C23
-:04017D00000000631B
-:04017E000000006419
-:04017F000000006418
-:040C40000000057437
+:04017C00000000641B
+:04017D000000003648
+:04017E000000003845
+:04017F000000003D3F
+:040C40000000040BA1
 :043FFE0000000500BA
 :00000001FF
processing file asmFiles/mult.asm (12 of 22)
processing file asmFiles/mult_procedure.asm (13 of 22)
processing file asmFiles/quicksort.asm (14 of 22)
processing file asmFiles/search.asm (15 of 22)
processing file asmFiles/slt.asm (16 of 22)
processing file asmFiles/test.branchjump1.asm (17 of 22)
processing file asmFiles/test.halt.asm (18 of 22)
processing file asmFiles/test.ldst1.asm (19 of 22)
processing file asmFiles/test.ldst2.asm (20 of 22)
processing file asmFiles/test.loadstore.asm (21 of 22)
processing file asmFiles/test.rtype.asm (22 of 22)
processing file asmFiles/mergesort.asm (1 of 1)
--- memsim.hex	2021-10-27 21:19:34.960480534 -0400
+++ memcpu.hex	2021-10-27 21:19:34.860476909 -0400
@@ -87,32 +87,32 @@
 :0400C4000000000731
 :0400C5000000000B2C
 :0400C6000000000C2A
-:0400C7000000001520
-:0400C800000000161E
-:0400C900000000161D
-:0400CA000000001919
-:0400CB000000001B16
-:0400CC000000001D13
-:0400CD00000000200F
-:0400CE00000000210D
-:0400CF000000002904
-:0400D00000000033F9
-:0400D10000000036F5
-:0400D20000000038F2
-:0400D3000000003DEC
-:0400D4000000003DEB
-:0400D5000000003EE9
-:0400D60000000042E4
-:0400D70000000047DE
-:0400D8000000004BD9
-:0400D9000000004DD6
-:0400DA0000000050D2
-:0400DB0000000051D0
-:0400DC0000000051CF
-:0400DD0000000053CC
-:0400DE000000005AC4
-:0400DF0000000063BA
-:0400E00000000064B8
+:0400C700000000161F
+:0400C8000000003301
+:0400C90000000051E2
+:0400CA000000005AD8
+:0400CB000000001918
+:0400CC000000002907
+:0400CD0000000050DF
+:0400CE000000001519
+:0400CF000000001617
+:0400D0000000001B11
+:0400D1000000003EED
+:0400D20000000047E3
+:0400D3000000004BDE
+:0400D4000000004DDB
+:0400D50000000051D6
+:0400D60000000053D3
+:0400D7000000001D08
+:0400D8000000002004
+:0400D9000000002102
+:0400DA000000003DE5
+:0400DB0000000042DF
+:0400DC0000000063BD
+:0400DD0000000064BB
+:0400DE0000000036E8
+:0400DF0000000038E5
+:0400E0000000003DDF
 :0400E1000000000714
 :0400E2000000000713
 :0400E3000000000712
@@ -128,23 +128,23 @@
 :0400ED000000001AF5
 :0400EE0000000020EE
 :0400EF0000000020ED
-:0400F00000000023E9
-:0400F10000000024E7
-:0400F20000000026E4
-:0400F30000000026E3
-:0400F40000000028E0
-:0400F5000000002FD8
-:0400F60000000030D6
-:0400F70000000032D3
-:0400F80000000041C3
-:0400F90000000042C1
-:0400FA0000000043BF
-:0400FB0000000044BD
-:0400FC0000000059A7
-:0400FD000000005BA4
-:0400FE000000005CA2
-:0400FF000000005CA1
-:040100000000006497
+:0400F0000000002FDD
+:0400F10000000030DB
+:0400F20000000041C9
+:0400F30000000042C7
+:0400F40000000059AF
+:0400F5000000005CAB
+:0400F60000000064A2
+:0400F70000000023E2
+:0400F80000000024E0
+:0400F90000000026DD
+:0400FA0000000026DC
+:0400FB0000000028D9
+:0400FC0000000032CE
+:0400FD0000000043BC
+:0400FE0000000044BA
+:0400FF000000005BA2
+:040100000000005C9F
 :0401400000000001BA
 :0401410000000003B7
 :0401420000000004B5
@@ -162,53 +162,53 @@
 :04014E000000000F9E
 :04014F00000000109C
 :04015000000000119A
-:040151000000001595
-:040152000000001693
-:040153000000001692
-:040154000000001790
-:04015500000000198D
-:040156000000001A8B
-:040157000000001B89
-:040158000000001D86
-:040159000000002082
-:04015A000000002081
-:04015B000000002080
-:04015C00000000217E
-:04015D00000000237B
-:04015E000000002479
-:04015F000000002676
-:040160000000002675
-:040161000000002872
-:040162000000002970
-:040163000000002F69
-:040164000000003067
-:040165000000003264
-:040166000000003362
-:04016700000000365E
-:04016800000000385B
-:040169000000003D55
-:04016A000000003D54
-:04016B000000003E52
-:04016C00000000414E
-:04016D00000000424C
+:040151000000001694
+:040152000000001792
+:040153000000001A8E
+:040154000000002087
+:040155000000002086
+:040156000000002F76
+:040157000000003074
+:040158000000003370
+:040159000000004161
+:04015A00000000425F
+:04015B00000000514F
+:04015C000000005946
+:04015D000000005A44
+:04015E000000001984
+:04015F000000002973
+:04016000000000504B
+:040161000000001585
+:040162000000001683
+:040163000000001B7D
+:040164000000003E59
+:04016500000000474F
+:040166000000004B4A
+:040167000000004D47
+:040168000000005142
+:04016900000000533F
+:04016A000000001D74
+:04016B000000002070
+:04016C00000000216E
+:04016D000000003D51
 :04016E00000000424B
-:04016F000000004349
-:040170000000004447
-:040171000000004743
-:040172000000004B3E
-:040173000000004D3B
-:040174000000005037
-:040175000000005135
-:040176000000005134
-:040177000000005331
-:04017800000000592A
-:040179000000005A28
+:04016F000000005C30
+:040170000000006328
+:040171000000006426
+:040172000000002366
+:040173000000002464
+:040174000000002661
+:040175000000002660
+:04017600000000285D
+:040177000000003252
+:040178000000004340
+:04017900000000443E
 :04017A000000005B26
 :04017B000000005C24
-:04017C000000005C23
-:04017D00000000631B
-:04017E000000006419
-:04017F000000006418
-:040C40000000057437
+:04017C00000000641B
+:04017D000000003648
+:04017E000000003845
+:04017F000000003D3F
+:040C40000000040BA1
 :043FFE0000000500BA
 :00000001FF
processing file asmFiles/mergesort.asm (1 of 1)
--- memsim.hex	2021-10-27 21:24:57.527172498 -0400
+++ memcpu.hex	2021-10-27 21:24:57.406168112 -0400
@@ -87,32 +87,32 @@
 :0400C4000000000731
 :0400C5000000000B2C
 :0400C6000000000C2A
-:0400C7000000001520
-:0400C800000000161E
-:0400C900000000161D
-:0400CA000000001919
-:0400CB000000001B16
-:0400CC000000001D13
-:0400CD00000000200F
-:0400CE00000000210D
-:0400CF000000002904
-:0400D00000000033F9
-:0400D10000000036F5
-:0400D20000000038F2
-:0400D3000000003DEC
-:0400D4000000003DEB
-:0400D5000000003EE9
-:0400D60000000042E4
-:0400D70000000047DE
-:0400D8000000004BD9
-:0400D9000000004DD6
-:0400DA0000000050D2
-:0400DB0000000051D0
-:0400DC0000000051CF
-:0400DD0000000053CC
-:0400DE000000005AC4
-:0400DF0000000063BA
-:0400E00000000064B8
+:0400C700000000161F
+:0400C8000000003301
+:0400C90000000051E2
+:0400CA000000005AD8
+:0400CB000000001918
+:0400CC000000002907
+:0400CD0000000050DF
+:0400CE000000001519
+:0400CF000000001617
+:0400D0000000001B11
+:0400D1000000003EED
+:0400D20000000047E3
+:0400D3000000004BDE
+:0400D4000000004DDB
+:0400D50000000051D6
+:0400D60000000053D3
+:0400D7000000001D08
+:0400D8000000002004
+:0400D9000000002102
+:0400DA000000003DE5
+:0400DB0000000042DF
+:0400DC0000000063BD
+:0400DD0000000064BB
+:0400DE0000000036E8
+:0400DF0000000038E5
+:0400E0000000003DDF
 :0400E1000000000714
 :0400E2000000000713
 :0400E3000000000712
@@ -128,23 +128,23 @@
 :0400ED000000001AF5
 :0400EE0000000020EE
 :0400EF0000000020ED
-:0400F00000000023E9
-:0400F10000000024E7
-:0400F20000000026E4
-:0400F30000000026E3
-:0400F40000000028E0
-:0400F5000000002FD8
-:0400F60000000030D6
-:0400F70000000032D3
-:0400F80000000041C3
-:0400F90000000042C1
-:0400FA0000000043BF
-:0400FB0000000044BD
-:0400FC0000000059A7
-:0400FD000000005BA4
-:0400FE000000005CA2
-:0400FF000000005CA1
-:040100000000006497
+:0400F0000000002FDD
+:0400F10000000030DB
+:0400F20000000041C9
+:0400F30000000042C7
+:0400F40000000059AF
+:0400F5000000005CAB
+:0400F60000000064A2
+:0400F70000000023E2
+:0400F80000000024E0
+:0400F90000000026DD
+:0400FA0000000026DC
+:0400FB0000000028D9
+:0400FC0000000032CE
+:0400FD0000000043BC
+:0400FE0000000044BA
+:0400FF000000005BA2
+:040100000000005C9F
 :0401400000000001BA
 :0401410000000003B7
 :0401420000000004B5
@@ -162,53 +162,53 @@
 :04014E000000000F9E
 :04014F00000000109C
 :04015000000000119A
-:040151000000001595
-:040152000000001693
-:040153000000001692
-:040154000000001790
-:04015500000000198D
-:040156000000001A8B
-:040157000000001B89
-:040158000000001D86
-:040159000000002082
-:04015A000000002081
-:04015B000000002080
-:04015C00000000217E
-:04015D00000000237B
-:04015E000000002479
-:04015F000000002676
-:040160000000002675
-:040161000000002872
-:040162000000002970
-:040163000000002F69
-:040164000000003067
-:040165000000003264
-:040166000000003362
-:04016700000000365E
-:04016800000000385B
-:040169000000003D55
-:04016A000000003D54
-:04016B000000003E52
-:04016C00000000414E
-:04016D00000000424C
+:040151000000001694
+:040152000000001792
+:040153000000001A8E
+:040154000000002087
+:040155000000002086
+:040156000000002F76
+:040157000000003074
+:040158000000003370
+:040159000000004161
+:04015A00000000425F
+:04015B00000000514F
+:04015C000000005946
+:04015D000000005A44
+:04015E000000001984
+:04015F000000002973
+:04016000000000504B
+:040161000000001585
+:040162000000001683
+:040163000000001B7D
+:040164000000003E59
+:04016500000000474F
+:040166000000004B4A
+:040167000000004D47
+:040168000000005142
+:04016900000000533F
+:04016A000000001D74
+:04016B000000002070
+:04016C00000000216E
+:04016D000000003D51
 :04016E00000000424B
-:04016F000000004349
-:040170000000004447
-:040171000000004743
-:040172000000004B3E
-:040173000000004D3B
-:040174000000005037
-:040175000000005135
-:040176000000005134
-:040177000000005331
-:04017800000000592A
-:040179000000005A28
+:04016F000000005C30
+:040170000000006328
+:040171000000006426
+:040172000000002366
+:040173000000002464
+:040174000000002661
+:040175000000002660
+:04017600000000285D
+:040177000000003252
+:040178000000004340
+:04017900000000443E
 :04017A000000005B26
 :04017B000000005C24
-:04017C000000005C23
-:04017D00000000631B
-:04017E000000006419
-:04017F000000006418
-:040C40000000057437
+:04017C00000000641B
+:04017D000000003648
+:04017E000000003845
+:04017F000000003D3F
+:040C40000000040BA1
 :043FFE0000000500BA
 :00000001FF
processing file asmFiles/branch.asm (1 of 22)
processing file asmFiles/cache_associate.asm (2 of 22)
processing file asmFiles/cache_ihit_before.asm (3 of 22)
processing file asmFiles/cache_ihit_dhit.asm (4 of 22)
processing file asmFiles/count_days.asm (5 of 22)
processing file asmFiles/fib.asm (6 of 22)
processing file asmFiles/hazard.control.asm (7 of 22)
processing file asmFiles/hazard.data.asm (8 of 22)
processing file asmFiles/hazard.structure.asm (9 of 22)
processing file asmFiles/jump.asm (10 of 22)
processing file asmFiles/mergesort.asm (11 of 22)
--- memsim.hex	2021-10-27 21:25:26.091207986 -0400
+++ memcpu.hex	2021-10-27 21:25:26.022205485 -0400
@@ -87,32 +87,32 @@
 :0400C4000000000731
 :0400C5000000000B2C
 :0400C6000000000C2A
-:0400C7000000001520
-:0400C800000000161E
-:0400C900000000161D
-:0400CA000000001919
-:0400CB000000001B16
-:0400CC000000001D13
-:0400CD00000000200F
-:0400CE00000000210D
-:0400CF000000002904
-:0400D00000000033F9
-:0400D10000000036F5
-:0400D20000000038F2
-:0400D3000000003DEC
-:0400D4000000003DEB
-:0400D5000000003EE9
-:0400D60000000042E4
-:0400D70000000047DE
-:0400D8000000004BD9
-:0400D9000000004DD6
-:0400DA0000000050D2
-:0400DB0000000051D0
-:0400DC0000000051CF
-:0400DD0000000053CC
-:0400DE000000005AC4
-:0400DF0000000063BA
-:0400E00000000064B8
+:0400C700000000161F
+:0400C8000000003301
+:0400C90000000051E2
+:0400CA000000005AD8
+:0400CB000000001918
+:0400CC000000002907
+:0400CD0000000050DF
+:0400CE000000001519
+:0400CF000000001617
+:0400D0000000001B11
+:0400D1000000003EED
+:0400D20000000047E3
+:0400D3000000004BDE
+:0400D4000000004DDB
+:0400D50000000051D6
+:0400D60000000053D3
+:0400D7000000001D08
+:0400D8000000002004
+:0400D9000000002102
+:0400DA000000003DE5
+:0400DB0000000042DF
+:0400DC0000000063BD
+:0400DD0000000064BB
+:0400DE0000000036E8
+:0400DF0000000038E5
+:0400E0000000003DDF
 :0400E1000000000714
 :0400E2000000000713
 :0400E3000000000712
@@ -128,23 +128,23 @@
 :0400ED000000001AF5
 :0400EE0000000020EE
 :0400EF0000000020ED
-:0400F00000000023E9
-:0400F10000000024E7
-:0400F20000000026E4
-:0400F30000000026E3
-:0400F40000000028E0
-:0400F5000000002FD8
-:0400F60000000030D6
-:0400F70000000032D3
-:0400F80000000041C3
-:0400F90000000042C1
-:0400FA0000000043BF
-:0400FB0000000044BD
-:0400FC0000000059A7
-:0400FD000000005BA4
-:0400FE000000005CA2
-:0400FF000000005CA1
-:040100000000006497
+:0400F0000000002FDD
+:0400F10000000030DB
+:0400F20000000041C9
+:0400F30000000042C7
+:0400F40000000059AF
+:0400F5000000005CAB
+:0400F60000000064A2
+:0400F70000000023E2
+:0400F80000000024E0
+:0400F90000000026DD
+:0400FA0000000026DC
+:0400FB0000000028D9
+:0400FC0000000032CE
+:0400FD0000000043BC
+:0400FE0000000044BA
+:0400FF000000005BA2
+:040100000000005C9F
 :0401400000000001BA
 :0401410000000003B7
 :0401420000000004B5
@@ -162,53 +162,53 @@
 :04014E000000000F9E
 :04014F00000000109C
 :04015000000000119A
-:040151000000001595
-:040152000000001693
-:040153000000001692
-:040154000000001790
-:04015500000000198D
-:040156000000001A8B
-:040157000000001B89
-:040158000000001D86
-:040159000000002082
-:04015A000000002081
-:04015B000000002080
-:04015C00000000217E
-:04015D00000000237B
-:04015E000000002479
-:04015F000000002676
-:040160000000002675
-:040161000000002872
-:040162000000002970
-:040163000000002F69
-:040164000000003067
-:040165000000003264
-:040166000000003362
-:04016700000000365E
-:04016800000000385B
-:040169000000003D55
-:04016A000000003D54
-:04016B000000003E52
-:04016C00000000414E
-:04016D00000000424C
+:040151000000001694
+:040152000000001792
+:040153000000001A8E
+:040154000000002087
+:040155000000002086
+:040156000000002F76
+:040157000000003074
+:040158000000003370
+:040159000000004161
+:04015A00000000425F
+:04015B00000000514F
+:04015C000000005946
+:04015D000000005A44
+:04015E000000001984
+:04015F000000002973
+:04016000000000504B
+:040161000000001585
+:040162000000001683
+:040163000000001B7D
+:040164000000003E59
+:04016500000000474F
+:040166000000004B4A
+:040167000000004D47
+:040168000000005142
+:04016900000000533F
+:04016A000000001D74
+:04016B000000002070
+:04016C00000000216E
+:04016D000000003D51
 :04016E00000000424B
-:04016F000000004349
-:040170000000004447
-:040171000000004743
-:040172000000004B3E
-:040173000000004D3B
-:040174000000005037
-:040175000000005135
-:040176000000005134
-:040177000000005331
-:04017800000000592A
-:040179000000005A28
+:04016F000000005C30
+:040170000000006328
+:040171000000006426
+:040172000000002366
+:040173000000002464
+:040174000000002661
+:040175000000002660
+:04017600000000285D
+:040177000000003252
+:040178000000004340
+:04017900000000443E
 :04017A000000005B26
 :04017B000000005C24
-:04017C000000005C23
-:04017D00000000631B
-:04017E000000006419
-:04017F000000006418
-:040C40000000057437
+:04017C00000000641B
+:04017D000000003648
+:04017E000000003845
+:04017F000000003D3F
+:040C40000000040BA1
 :043FFE0000000500BA
 :00000001FF
processing file asmFiles/mult.asm (12 of 22)
processing file asmFiles/mult_procedure.asm (13 of 22)
processing file asmFiles/quicksort.asm (14 of 22)
processing file asmFiles/search.asm (15 of 22)
processing file asmFiles/slt.asm (16 of 22)
processing file asmFiles/test.branchjump1.asm (17 of 22)
processing file asmFiles/test.halt.asm (18 of 22)
processing file asmFiles/test.ldst1.asm (19 of 22)
processing file asmFiles/test.ldst2.asm (20 of 22)
processing file asmFiles/test.loadstore.asm (21 of 22)
processing file asmFiles/test.rtype.asm (22 of 22)
processing file asmFiles/mergesort.asm (1 of 1)
--- memsim.hex	2021-10-27 21:26:57.435519351 -0400
+++ memcpu.hex	2021-10-27 21:26:57.335515725 -0400
@@ -87,32 +87,32 @@
 :0400C4000000000731
 :0400C5000000000B2C
 :0400C6000000000C2A
-:0400C7000000001520
-:0400C800000000161E
-:0400C900000000161D
-:0400CA000000001919
-:0400CB000000001B16
-:0400CC000000001D13
-:0400CD00000000200F
-:0400CE00000000210D
-:0400CF000000002904
-:0400D00000000033F9
-:0400D10000000036F5
-:0400D20000000038F2
-:0400D3000000003DEC
-:0400D4000000003DEB
-:0400D5000000003EE9
-:0400D60000000042E4
-:0400D70000000047DE
-:0400D8000000004BD9
-:0400D9000000004DD6
-:0400DA0000000050D2
-:0400DB0000000051D0
-:0400DC0000000051CF
-:0400DD0000000053CC
-:0400DE000000005AC4
-:0400DF0000000063BA
-:0400E00000000064B8
+:0400C700000000161F
+:0400C8000000003301
+:0400C90000000051E2
+:0400CA000000005AD8
+:0400CB000000001918
+:0400CC000000002907
+:0400CD0000000050DF
+:0400CE000000001519
+:0400CF000000001617
+:0400D0000000001B11
+:0400D1000000003EED
+:0400D20000000047E3
+:0400D3000000004BDE
+:0400D4000000004DDB
+:0400D50000000051D6
+:0400D60000000053D3
+:0400D7000000001D08
+:0400D8000000002004
+:0400D9000000002102
+:0400DA000000003DE5
+:0400DB0000000042DF
+:0400DC0000000063BD
+:0400DD0000000064BB
+:0400DE0000000036E8
+:0400DF0000000038E5
+:0400E0000000003DDF
 :0400E1000000000714
 :0400E2000000000713
 :0400E3000000000712
@@ -128,23 +128,23 @@
 :0400ED000000001AF5
 :0400EE0000000020EE
 :0400EF0000000020ED
-:0400F00000000023E9
-:0400F10000000024E7
-:0400F20000000026E4
-:0400F30000000026E3
-:0400F40000000028E0
-:0400F5000000002FD8
-:0400F60000000030D6
-:0400F70000000032D3
-:0400F80000000041C3
-:0400F90000000042C1
-:0400FA0000000043BF
-:0400FB0000000044BD
-:0400FC0000000059A7
-:0400FD000000005BA4
-:0400FE000000005CA2
-:0400FF000000005CA1
-:040100000000006497
+:0400F0000000002FDD
+:0400F10000000030DB
+:0400F20000000041C9
+:0400F30000000042C7
+:0400F40000000059AF
+:0400F5000000005CAB
+:0400F60000000064A2
+:0400F70000000023E2
+:0400F80000000024E0
+:0400F90000000026DD
+:0400FA0000000026DC
+:0400FB0000000028D9
+:0400FC0000000032CE
+:0400FD0000000043BC
+:0400FE0000000044BA
+:0400FF000000005BA2
+:040100000000005C9F
 :0401400000000001BA
 :0401410000000003B7
 :0401420000000004B5
@@ -162,53 +162,53 @@
 :04014E000000000F9E
 :04014F00000000109C
 :04015000000000119A
-:040151000000001595
-:040152000000001693
-:040153000000001692
-:040154000000001790
-:04015500000000198D
-:040156000000001A8B
-:040157000000001B89
-:040158000000001D86
-:040159000000002082
-:04015A000000002081
-:04015B000000002080
-:04015C00000000217E
-:04015D00000000237B
-:04015E000000002479
-:04015F000000002676
-:040160000000002675
-:040161000000002872
-:040162000000002970
-:040163000000002F69
-:040164000000003067
-:040165000000003264
-:040166000000003362
-:04016700000000365E
-:04016800000000385B
-:040169000000003D55
-:04016A000000003D54
-:04016B000000003E52
-:04016C00000000414E
-:04016D00000000424C
+:040151000000001694
+:040152000000001792
+:040153000000001A8E
+:040154000000002087
+:040155000000002086
+:040156000000002F76
+:040157000000003074
+:040158000000003370
+:040159000000004161
+:04015A00000000425F
+:04015B00000000514F
+:04015C000000005946
+:04015D000000005A44
+:04015E000000001984
+:04015F000000002973
+:04016000000000504B
+:040161000000001585
+:040162000000001683
+:040163000000001B7D
+:040164000000003E59
+:04016500000000474F
+:040166000000004B4A
+:040167000000004D47
+:040168000000005142
+:04016900000000533F
+:04016A000000001D74
+:04016B000000002070
+:04016C00000000216E
+:04016D000000003D51
 :04016E00000000424B
-:04016F000000004349
-:040170000000004447
-:040171000000004743
-:040172000000004B3E
-:040173000000004D3B
-:040174000000005037
-:040175000000005135
-:040176000000005134
-:040177000000005331
-:04017800000000592A
-:040179000000005A28
+:04016F000000005C30
+:040170000000006328
+:040171000000006426
+:040172000000002366
+:040173000000002464
+:040174000000002661
+:040175000000002660
+:04017600000000285D
+:040177000000003252
+:040178000000004340
+:04017900000000443E
 :04017A000000005B26
 :04017B000000005C24
-:04017C000000005C23
-:04017D00000000631B
-:04017E000000006419
-:04017F000000006418
-:040C40000000057437
+:04017C00000000641B
+:04017D000000003648
+:04017E000000003845
+:04017F000000003D3F
+:040C40000000040BA1
 :043FFE0000000500BA
 :00000001FF
processing file asmFiles/branch.asm (1 of 22)
processing file asmFiles/cache_associate.asm (2 of 22)
processing file asmFiles/cache_ihit_before.asm (3 of 22)
processing file asmFiles/cache_ihit_dhit.asm (4 of 22)
processing file asmFiles/count_days.asm (5 of 22)
processing file asmFiles/fib.asm (6 of 22)
processing file asmFiles/hazard.control.asm (7 of 22)
processing file asmFiles/hazard.data.asm (8 of 22)
processing file asmFiles/hazard.structure.asm (9 of 22)
processing file asmFiles/jump.asm (10 of 22)
processing file asmFiles/mergesort.asm (11 of 22)
--- memsim.hex	2021-10-27 21:30:43.117700659 -0400
+++ memcpu.hex	2021-10-27 21:30:43.011696816 -0400
@@ -87,32 +87,32 @@
 :0400C4000000000731
 :0400C5000000000B2C
 :0400C6000000000C2A
-:0400C7000000001520
-:0400C800000000161E
-:0400C900000000161D
-:0400CA000000001919
-:0400CB000000001B16
-:0400CC000000001D13
-:0400CD00000000200F
-:0400CE00000000210D
-:0400CF000000002904
-:0400D00000000033F9
-:0400D10000000036F5
-:0400D20000000038F2
-:0400D3000000003DEC
-:0400D4000000003DEB
-:0400D5000000003EE9
-:0400D60000000042E4
-:0400D70000000047DE
-:0400D8000000004BD9
-:0400D9000000004DD6
-:0400DA0000000050D2
-:0400DB0000000051D0
-:0400DC0000000051CF
-:0400DD0000000053CC
-:0400DE000000005AC4
-:0400DF0000000063BA
-:0400E00000000064B8
+:0400C700000000161F
+:0400C8000000003301
+:0400C90000000051E2
+:0400CA000000005AD8
+:0400CB000000001918
+:0400CC000000002907
+:0400CD0000000050DF
+:0400CE000000001519
+:0400CF000000001617
+:0400D0000000001B11
+:0400D1000000003EED
+:0400D20000000047E3
+:0400D3000000004BDE
+:0400D4000000004DDB
+:0400D50000000051D6
+:0400D60000000053D3
+:0400D7000000001D08
+:0400D8000000002004
+:0400D9000000002102
+:0400DA000000003DE5
+:0400DB0000000042DF
+:0400DC0000000063BD
+:0400DD0000000064BB
+:0400DE0000000036E8
+:0400DF0000000038E5
+:0400E0000000003DDF
 :0400E1000000000714
 :0400E2000000000713
 :0400E3000000000712
@@ -128,23 +128,23 @@
 :0400ED000000001AF5
 :0400EE0000000020EE
 :0400EF0000000020ED
-:0400F00000000023E9
-:0400F10000000024E7
-:0400F20000000026E4
-:0400F30000000026E3
-:0400F40000000028E0
-:0400F5000000002FD8
-:0400F60000000030D6
-:0400F70000000032D3
-:0400F80000000041C3
-:0400F90000000042C1
-:0400FA0000000043BF
-:0400FB0000000044BD
-:0400FC0000000059A7
-:0400FD000000005BA4
-:0400FE000000005CA2
-:0400FF000000005CA1
-:040100000000006497
+:0400F0000000002FDD
+:0400F10000000030DB
+:0400F20000000041C9
+:0400F30000000042C7
+:0400F40000000059AF
+:0400F5000000005CAB
+:0400F60000000064A2
+:0400F70000000023E2
+:0400F80000000024E0
+:0400F90000000026DD
+:0400FA0000000026DC
+:0400FB0000000028D9
+:0400FC0000000032CE
+:0400FD0000000043BC
+:0400FE0000000044BA
+:0400FF000000005BA2
+:040100000000005C9F
 :0401400000000001BA
 :0401410000000003B7
 :0401420000000004B5
@@ -162,53 +162,53 @@
 :04014E000000000F9E
 :04014F00000000109C
 :04015000000000119A
-:040151000000001595
-:040152000000001693
-:040153000000001692
-:040154000000001790
-:04015500000000198D
-:040156000000001A8B
-:040157000000001B89
-:040158000000001D86
-:040159000000002082
-:04015A000000002081
-:04015B000000002080
-:04015C00000000217E
-:04015D00000000237B
-:04015E000000002479
-:04015F000000002676
-:040160000000002675
-:040161000000002872
-:040162000000002970
-:040163000000002F69
-:040164000000003067
-:040165000000003264
-:040166000000003362
-:04016700000000365E
-:04016800000000385B
-:040169000000003D55
-:04016A000000003D54
-:04016B000000003E52
-:04016C00000000414E
-:04016D00000000424C
+:040151000000001694
+:040152000000001792
+:040153000000001A8E
+:040154000000002087
+:040155000000002086
+:040156000000002F76
+:040157000000003074
+:040158000000003370
+:040159000000004161
+:04015A00000000425F
+:04015B00000000514F
+:04015C000000005946
+:04015D000000005A44
+:04015E000000001984
+:04015F000000002973
+:04016000000000504B
+:040161000000001585
+:040162000000001683
+:040163000000001B7D
+:040164000000003E59
+:04016500000000474F
+:040166000000004B4A
+:040167000000004D47
+:040168000000005142
+:04016900000000533F
+:04016A000000001D74
+:04016B000000002070
+:04016C00000000216E
+:04016D000000003D51
 :04016E00000000424B
-:04016F000000004349
-:040170000000004447
-:040171000000004743
-:040172000000004B3E
-:040173000000004D3B
-:040174000000005037
-:040175000000005135
-:040176000000005134
-:040177000000005331
-:04017800000000592A
-:040179000000005A28
+:04016F000000005C30
+:040170000000006328
+:040171000000006426
+:040172000000002366
+:040173000000002464
+:040174000000002661
+:040175000000002660
+:04017600000000285D
+:040177000000003252
+:040178000000004340
+:04017900000000443E
 :04017A000000005B26
 :04017B000000005C24
-:04017C000000005C23
-:04017D00000000631B
-:04017E000000006419
-:04017F000000006418
-:040C40000000057437
+:04017C00000000641B
+:04017D000000003648
+:04017E000000003845
+:04017F000000003D3F
+:040C40000000040BA1
 :043FFE0000000500BA
 :00000001FF
processing file asmFiles/mult.asm (12 of 22)
processing file asmFiles/mult_procedure.asm (13 of 22)
processing file asmFiles/quicksort.asm (14 of 22)
processing file asmFiles/search.asm (15 of 22)
processing file asmFiles/slt.asm (16 of 22)
processing file asmFiles/test.branchjump1.asm (17 of 22)
processing file asmFiles/test.halt.asm (18 of 22)
processing file asmFiles/test.ldst1.asm (19 of 22)
processing file asmFiles/test.ldst2.asm (20 of 22)
processing file asmFiles/test.loadstore.asm (21 of 22)
processing file asmFiles/test.rtype.asm (22 of 22)
processing file asmFiles/mergesort.asm (1 of 1)
--- memsim.hex	2021-10-27 21:31:24.774210768 -0400
+++ memcpu.hex	2021-10-27 21:31:24.657206526 -0400
@@ -87,32 +87,32 @@
 :0400C4000000000731
 :0400C5000000000B2C
 :0400C6000000000C2A
-:0400C7000000001520
-:0400C800000000161E
-:0400C900000000161D
-:0400CA000000001919
-:0400CB000000001B16
-:0400CC000000001D13
-:0400CD00000000200F
-:0400CE00000000210D
-:0400CF000000002904
-:0400D00000000033F9
-:0400D10000000036F5
-:0400D20000000038F2
-:0400D3000000003DEC
-:0400D4000000003DEB
-:0400D5000000003EE9
-:0400D60000000042E4
-:0400D70000000047DE
-:0400D8000000004BD9
-:0400D9000000004DD6
-:0400DA0000000050D2
-:0400DB0000000051D0
-:0400DC0000000051CF
-:0400DD0000000053CC
-:0400DE000000005AC4
-:0400DF0000000063BA
-:0400E00000000064B8
+:0400C700000000161F
+:0400C8000000003301
+:0400C90000000051E2
+:0400CA000000005AD8
+:0400CB000000001918
+:0400CC000000002907
+:0400CD0000000050DF
+:0400CE000000001519
+:0400CF000000001617
+:0400D0000000001B11
+:0400D1000000003EED
+:0400D20000000047E3
+:0400D3000000004BDE
+:0400D4000000004DDB
+:0400D50000000051D6
+:0400D60000000053D3
+:0400D7000000001D08
+:0400D8000000002004
+:0400D9000000002102
+:0400DA000000003DE5
+:0400DB0000000042DF
+:0400DC0000000063BD
+:0400DD0000000064BB
+:0400DE0000000036E8
+:0400DF0000000038E5
+:0400E0000000003DDF
 :0400E1000000000714
 :0400E2000000000713
 :0400E3000000000712
@@ -128,23 +128,23 @@
 :0400ED000000001AF5
 :0400EE0000000020EE
 :0400EF0000000020ED
-:0400F00000000023E9
-:0400F10000000024E7
-:0400F20000000026E4
-:0400F30000000026E3
-:0400F40000000028E0
-:0400F5000000002FD8
-:0400F60000000030D6
-:0400F70000000032D3
-:0400F80000000041C3
-:0400F90000000042C1
-:0400FA0000000043BF
-:0400FB0000000044BD
-:0400FC0000000059A7
-:0400FD000000005BA4
-:0400FE000000005CA2
-:0400FF000000005CA1
-:040100000000006497
+:0400F0000000002FDD
+:0400F10000000030DB
+:0400F20000000041C9
+:0400F30000000042C7
+:0400F40000000059AF
+:0400F5000000005CAB
+:0400F60000000064A2
+:0400F70000000023E2
+:0400F80000000024E0
+:0400F90000000026DD
+:0400FA0000000026DC
+:0400FB0000000028D9
+:0400FC0000000032CE
+:0400FD0000000043BC
+:0400FE0000000044BA
+:0400FF000000005BA2
+:040100000000005C9F
 :0401400000000001BA
 :0401410000000003B7
 :0401420000000004B5
@@ -162,53 +162,53 @@
 :04014E000000000F9E
 :04014F00000000109C
 :04015000000000119A
-:040151000000001595
-:040152000000001693
-:040153000000001692
-:040154000000001790
-:04015500000000198D
-:040156000000001A8B
-:040157000000001B89
-:040158000000001D86
-:040159000000002082
-:04015A000000002081
-:04015B000000002080
-:04015C00000000217E
-:04015D00000000237B
-:04015E000000002479
-:04015F000000002676
-:040160000000002675
-:040161000000002872
-:040162000000002970
-:040163000000002F69
-:040164000000003067
-:040165000000003264
-:040166000000003362
-:04016700000000365E
-:04016800000000385B
-:040169000000003D55
-:04016A000000003D54
-:04016B000000003E52
-:04016C00000000414E
-:04016D00000000424C
+:040151000000001694
+:040152000000001792
+:040153000000001A8E
+:040154000000002087
+:040155000000002086
+:040156000000002F76
+:040157000000003074
+:040158000000003370
+:040159000000004161
+:04015A00000000425F
+:04015B00000000514F
+:04015C000000005946
+:04015D000000005A44
+:04015E000000001984
+:04015F000000002973
+:04016000000000504B
+:040161000000001585
+:040162000000001683
+:040163000000001B7D
+:040164000000003E59
+:04016500000000474F
+:040166000000004B4A
+:040167000000004D47
+:040168000000005142
+:04016900000000533F
+:04016A000000001D74
+:04016B000000002070
+:04016C00000000216E
+:04016D000000003D51
 :04016E00000000424B
-:04016F000000004349
-:040170000000004447
-:040171000000004743
-:040172000000004B3E
-:040173000000004D3B
-:040174000000005037
-:040175000000005135
-:040176000000005134
-:040177000000005331
-:04017800000000592A
-:040179000000005A28
+:04016F000000005C30
+:040170000000006328
+:040171000000006426
+:040172000000002366
+:040173000000002464
+:040174000000002661
+:040175000000002660
+:04017600000000285D
+:040177000000003252
+:040178000000004340
+:04017900000000443E
 :04017A000000005B26
 :04017B000000005C24
-:04017C000000005C23
-:04017D00000000631B
-:04017E000000006419
-:04017F000000006418
-:040C40000000057437
+:04017C00000000641B
+:04017D000000003648
+:04017E000000003845
+:04017F000000003D3F
+:040C40000000040BA1
 :043FFE0000000500BA
 :00000001FF
processing file asmFiles/mergesort.asm (1 of 1)
--- memsim.hex	2021-10-27 21:33:46.907363302 -0400
+++ memcpu.hex	2021-10-27 21:33:46.784358843 -0400
@@ -87,32 +87,32 @@
 :0400C4000000000731
 :0400C5000000000B2C
 :0400C6000000000C2A
-:0400C7000000001520
-:0400C800000000161E
-:0400C900000000161D
-:0400CA000000001919
-:0400CB000000001B16
-:0400CC000000001D13
-:0400CD00000000200F
-:0400CE00000000210D
-:0400CF000000002904
-:0400D00000000033F9
-:0400D10000000036F5
-:0400D20000000038F2
-:0400D3000000003DEC
-:0400D4000000003DEB
-:0400D5000000003EE9
-:0400D60000000042E4
-:0400D70000000047DE
-:0400D8000000004BD9
-:0400D9000000004DD6
-:0400DA0000000050D2
-:0400DB0000000051D0
-:0400DC0000000051CF
-:0400DD0000000053CC
-:0400DE000000005AC4
-:0400DF0000000063BA
-:0400E00000000064B8
+:0400C700000000161F
+:0400C8000000003301
+:0400C90000000051E2
+:0400CA000000005AD8
+:0400CB000000001918
+:0400CC000000002907
+:0400CD0000000050DF
+:0400CE000000001519
+:0400CF000000001617
+:0400D0000000001B11
+:0400D1000000003EED
+:0400D20000000047E3
+:0400D3000000004BDE
+:0400D4000000004DDB
+:0400D50000000051D6
+:0400D60000000053D3
+:0400D7000000001D08
+:0400D8000000002004
+:0400D9000000002102
+:0400DA000000003DE5
+:0400DB0000000042DF
+:0400DC0000000063BD
+:0400DD0000000064BB
+:0400DE0000000036E8
+:0400DF0000000038E5
+:0400E0000000003DDF
 :0400E1000000000714
 :0400E2000000000713
 :0400E3000000000712
@@ -128,23 +128,23 @@
 :0400ED000000001AF5
 :0400EE0000000020EE
 :0400EF0000000020ED
-:0400F00000000023E9
-:0400F10000000024E7
-:0400F20000000026E4
-:0400F30000000026E3
-:0400F40000000028E0
-:0400F5000000002FD8
-:0400F60000000030D6
-:0400F70000000032D3
-:0400F80000000041C3
-:0400F90000000042C1
-:0400FA0000000043BF
-:0400FB0000000044BD
-:0400FC0000000059A7
-:0400FD000000005BA4
-:0400FE000000005CA2
-:0400FF000000005CA1
-:040100000000006497
+:0400F0000000002FDD
+:0400F10000000030DB
+:0400F20000000041C9
+:0400F30000000042C7
+:0400F40000000059AF
+:0400F5000000005CAB
+:0400F60000000064A2
+:0400F70000000023E2
+:0400F80000000024E0
+:0400F90000000026DD
+:0400FA0000000026DC
+:0400FB0000000028D9
+:0400FC0000000032CE
+:0400FD0000000043BC
+:0400FE0000000044BA
+:0400FF000000005BA2
+:040100000000005C9F
 :0401400000000001BA
 :0401410000000003B7
 :0401420000000004B5
@@ -162,53 +162,39 @@
 :04014E000000000F9E
 :04014F00000000109C
 :04015000000000119A
-:040151000000001595
-:040152000000001693
-:040153000000001692
-:040154000000001790
-:04015500000000198D
-:040156000000001A8B
-:040157000000001B89
-:040158000000001D86
-:040159000000002082
-:04015A000000002081
-:04015B000000002080
-:04015C00000000217E
-:04015D00000000237B
-:04015E000000002479
-:04015F000000002676
-:040160000000002675
-:040161000000002872
-:040162000000002970
-:040163000000002F69
-:040164000000003067
-:040165000000003264
-:040166000000003362
-:04016700000000365E
-:04016800000000385B
-:040169000000003D55
-:04016A000000003D54
-:04016B000000003E52
-:04016C00000000414E
-:04016D00000000424C
+:040151000000001694
+:040152000000001792
+:040153000000001A8E
+:040154000000002087
+:040155000000002086
+:040156000000002F76
+:040157000000003074
+:040158000000003370
+:040159000000004161
+:04015A00000000425F
+:04015B00000000514F
+:04015C000000005946
+:04015D000000005A44
+:04015E000000001984
+:04015F000000002973
+:04016000000000504B
+:040161000000001585
+:040162000000001683
+:040163000000001B7D
+:040164000000003E59
+:04016500000000474F
+:040166000000004B4A
+:040167000000004D47
+:040168000000005142
+:04016900000000533F
+:04016A000000001D74
+:04016B000000002070
+:04016C00000000216E
+:04016D000000003D51
 :04016E00000000424B
-:04016F000000004349
-:040170000000004447
-:040171000000004743
-:040172000000004B3E
-:040173000000004D3B
-:040174000000005037
-:040175000000005135
-:040176000000005134
-:040177000000005331
-:04017800000000592A
-:040179000000005A28
-:04017A000000005B26
-:04017B000000005C24
-:04017C000000005C23
-:04017D00000000631B
-:04017E000000006419
-:04017F000000006418
-:040C40000000057437
+:04016F000000005C30
+:040170000000006328
+:040171000000006426
+:040C40000000040BA1
 :043FFE0000000500BA
 :00000001FF
processing file asmFiles/branch.asm (1 of 22)
processing file asmFiles/cache_associate.asm (2 of 22)
--- memsim.hex	2021-10-27 21:41:51.830938552 -0400
+++ memcpu.hex	2021-10-27 21:41:51.734935073 -0400
@@ -7,5 +7,6 @@
 :04000600FFFFFFFFFA
 :0403FF0000001234B4
 :04043F00000034562F
+:040C400000000002AE
 :043C3F000000234519
 :00000001FF
processing file asmFiles/cache_ihit_before.asm (3 of 22)
--- memsim.hex	2021-10-27 21:41:53.422996253 -0400
+++ memcpu.hex	2021-10-27 21:41:53.325992738 -0400
@@ -10,8 +10,9 @@
 :0400090014A0FFFA46
 :04000A00AD04000041
 :04000B00FFFFFFFFF5
-:04003C0000000002BE
-:04003D0000000001BE
-:04003F0000000003BA
-:040C400000000005AB
+:04003C0000000003BD
+:04003D000000BEEF12
+:04003E000000123478
+:0400400000000003B9
+:040C400000000003AD
 :00000001FF
processing file asmFiles/cache_ihit_dhit.asm (4 of 22)
processing file asmFiles/branch.asm (1 of 22)
processing file asmFiles/cache_associate.asm (2 of 22)
processing file asmFiles/cache_ihit_before.asm (3 of 22)
processing file asmFiles/cache_ihit_dhit.asm (4 of 22)
processing file asmFiles/count_days.asm (5 of 22)
processing file asmFiles/fib.asm (6 of 22)
processing file asmFiles/branch.asm (1 of 22)
processing file asmFiles/cache_associate.asm (2 of 22)
--- memsim.hex	2021-10-27 21:42:48.033975514 -0400
+++ memcpu.hex	2021-10-27 21:42:47.964973014 -0400
@@ -7,5 +7,6 @@
 :04000600FFFFFFFFFA
 :0403FF0000001234B4
 :04043F00000034562F
+:040C400000000001AF
 :043C3F000000234519
 :00000001FF
processing file asmFiles/cache_ihit_before.asm (3 of 22)
--- memsim.hex	2021-10-27 21:42:49.509028975 -0400
+++ memcpu.hex	2021-10-27 21:42:49.439026438 -0400
@@ -10,8 +10,7 @@
 :0400090014A0FFFA46
 :04000A00AD04000041
 :04000B00FFFFFFFFF5
-:04003C0000000002BE
-:04003D0000000001BE
-:04003F0000000003BA
-:040C400000000005AB
+:04003C0000000001BF
+:04003E0000000003BB
+:040C400000000004AC
 :00000001FF
processing file asmFiles/cache_ihit_dhit.asm (4 of 22)
processing file asmFiles/branch.asm (1 of 22)
processing file asmFiles/cache_associate.asm (2 of 22)
processing file asmFiles/cache_ihit_before.asm (3 of 22)
processing file asmFiles/cache_ihit_dhit.asm (4 of 22)
processing file asmFiles/count_days.asm (5 of 22)
processing file asmFiles/fib.asm (1 of 1)
processing file asmFiles/search.asm (1 of 1)
processing file asmFiles/mult.asm (1 of 2)
processing file asmFiles/mult_procedure.asm (2 of 2)
